snmp/tests/test_core_vs_python_e2e.py

```diff
@@ -0,0 +1,386 @@
+
+
+def test_e2e_profile_hpe_proliant(dd_agent_check):
+    config = common.generate_container_profile_config('hpe-proliant')
+    assert_python_vs_core(dd_agent_check, config)
+
+
+def test_e2e_profile_idrac(dd_agent_check):
+    config = common.generate_container_profile_config('idrac')
+    assert_python_vs_core(dd_agent_check, config)
+
+
+def test_e2e_profile_isilon(dd_agent_check):
+    config = common.generate_container_profile_config('isilon')
+    assert_python_vs_core(dd_agent_check, config)
+
+
+def test_e2e_profile_meraki_cloud_controller(dd_agent_check):
+    config = common.generate_container_profile_config('meraki-cloud-controller')
+    assert_python_vs_core(dd_agent_check, config)
+
+
+def test_e2e_profile_netapp(dd_agent_check):
+    config = common.generate_container_profile_config('netapp')
+    assert_python_vs_core(dd_agent_check, config)
+
+
+def test_e2e_profile_palo_alto(dd_agent_check):
+    config = common.generate_container_profile_config('palo-alto')
+    assert_python_vs_core(dd_agent_check, config)
+
+
+def assert_python_vs_core(dd_agent_check, config, expected_total_count=None, metrics_to_skip=None):
+    python_config = deepcopy(config)
+    python_config['init_config']['loader'] = 'python'
+    core_config = deepcopy(config)
+    core_config['init_config']['loader'] = 'core'
+    aggregator = dd_agent_check(python_config, rate=True)
+    expected_metrics = defaultdict(list)
+    metrics_to_skip = metrics_to_skip or []
+    for _, metrics in aggregator._metrics.items():
+        for stub in metrics:
+            if stub.name in metrics_to_skip:
+                continue
+            stub = normalize_stub_metric(stub)
+            expected_metrics[(stub.name, stub.type, tuple(sorted(stub.tags)))].append(stub)
+
+    expected_sc = defaultdict(list)
+    for _, service_checks in aggregator._service_checks.items():
+        for stub in service_checks:
+            expected_sc[(stub.name, stub.status, tuple(sorted(stub.tags)), stub.message)].append(stub)
+
+    total_count_python = sum(len(stubs) for stubs in expected_metrics.values())
+
+    aggregator.reset()
+    aggregator = dd_agent_check(core_config, rate=True)
+
+    aggregator_metrics = aggregator._metrics
+    aggregator._metrics = defaultdict(list)
+    for metric_name in aggregator_metrics:
+        for stub in aggregator_metrics[metric_name]:
+            if stub.name in metrics_to_skip:
+                continue
+            aggregator._metrics[metric_name].append(normalize_stub_metric(stub))
+
+    actual_metrics = defaultdict(list)
+    for _, metrics in aggregator._metrics.items():
+        for metric in metrics:
+            actual_metrics[(metric.name, metric.type, tuple(sorted(metric.tags)))].append(metric)
+
+    print("Python metrics not found in Corecheck metrics:")
+    for key in sorted(expected_metrics):
+        if key not in actual_metrics:
+            print("\t{}".format(key))
+
+    print("Corecheck metrics not found in Python metrics:")
```

Comment:
I'm missing something here: are you testing core vs python, or core vs stub?

Reply:
Seen now, can you rename `expected` to `python` and `actual` to `core` ? I would reserve `expected` to something static, here both values can be `wrong`, and to really understand who's missing something is to test both against static stub values. Does this make sense?

Reply:
I initialled started the e2e mainly for testing core. But make sense to make it more python vs core.

Fixed here: https://github.com/DataDog/integrations-core/pull/8708/commits/7bc1d3506aa3c32e0b9d91eddb3473134f62b59a